src/ui/layout/style.scss

```diff
@@ -0,0 +1,168 @@
+$spacings: 5;
+$directions: "y", "x", "t", "b", "r", "l";
+
+@each $direction in $directions {
+  @for $i from 1 through $spacings {
+    .p#{$direction}-#{$i} {
+      @if ($direction== "y") or ($direction== "t") {
+        padding-top: #{$i}rem;
+      }
+
+      @if ($direction== "y") or ($direction== "b") {
+        padding-bottom: #{$i}rem;
+      }
+
+      @if ($direction== "x") or ($direction== "r") {
+        padding-right: #{$i}rem;
+      }
+
+      @if ($direction== "x") or ($direction== "l") {
+        padding-left: #{$i}rem;
+      }
+    }
+  }
+}
+
+@each $direction in $directions {
+  @for $i from 1 through $spacings {
+    .m#{$direction}-#{$i} {
+      @if ($direction== "y") or ($direction== "t") {
+        margin-top: #{$i}rem;
```

Comment:
not too familiar with common practices for CSS, but would this mean that the `Layout` component is restricted to `rem`?

Reply:
Yeah, `rem` is a flexible unit which essential converts pixel values based on the browser's font size compared to using `px` which is absolute.

I wanted to have this specific styling so that we can have a consistent spacing usage across the components without needing to add custom margin/paddings (e.g. `1.2rem`) and throw out consistency.